internal/cluster/transition.go

```diff
@@ -482,6 +483,21 @@ func (th *transitionHandler) PostRefreshCluster(reason string) stateswitch.PostT
 				reason, params.eventHandler, extra...)
 		}
 
+		//report cluster install duration metrics in case of error. Cancel and Installed cases are
+		//treated separately in CancelInstallation and CompleteInstallation respectively
+		if sCluster.srcState != swag.StringValue(sCluster.cluster.Status) {
+			if reason == statusInfoError {
+				params.metricApi.ClusterInstallationFinished(params.ctx, "error", sCluster.srcState,
+					sCluster.cluster.OpenshiftVersion, *sCluster.cluster.ID, sCluster.cluster.EmailDomain,
+					sCluster.cluster.InstallStartedAt)
+			}
+			if reason == statusInfoInstallingPendingUserAction {
+				params.metricApi.ClusterInstallationFinished(params.ctx, "pending-user-action", sCluster.srcState,
```

Comment:
Currently, we have only 3 installation results: done, error & canceled. After this code, we will see also pending-user-action result - Not sure it's correct.

Reply:
This is a stopping point for the installation. We really can not measure the installation time including the user's manual intervention